### DISCUSSION TITLE
Dropping support for EOL Python 3.6

### DISCUSSION THREAD

**Discussion by ofek** [encode/httpx#2136] [Ideas] [answered]
Python 3.6 became EOL at the end of 2021, see: https://endoflife.date/python

Build backends like `setuptools` have already dropped support. This will not break anything because `pip` only looks for version candidates that satisfy the `requires-python` field. Here are stats for the curious: https://pypistats.org/packages/httpx

https://github.com/encode/httpx/pull/2097 implements this, cc @florimondmanca

**Comment by ofek** [comment]
@tomchristie Hello again! What do you think about this?

**Reply by lovelydinosaur** [reply]
🤷‍♂️  - I don't know.
What do other @encode/maintainers think?

**Reply by adriangb** [reply]
I would be in favor of dropping it for httpx.

**Reply by michaeloliverx** [reply]
> All in favor, say aye

Aye!

**Reply by Kludex** [reply]
I think the other projects are deciding this for us. As they are moving forward.

Maybe we should set up a policy of version support...

**Reply by florimondmanca** [reply]
I would be in favor, first because 3.6 is now EOL, but also because download stats show it is now among the least used versions (although such stats are always uncertain) - seems to have been for at least 6mo now:

https://pypistats.org/packages/httpx

- 3.6: ~5k/day
- 3.7: ~70k/day
- 3.8: ~70k/day
- 3.9: ~45k/day
- 3.10: ~10k/day
- 3.11: negligible
- "Null" (???): ~50k/day

Edit: oh ofek had already pointed that out in initial post. Well here are the numbers!

**Reply by ahopkins** [reply]
I will throw in my opinion and also say this is a nice, and worthwhile change. Although, looks like the PR was already merged :laughing: so I am late to the party.

**Comment by ofek** [comment, marked as answer]
Merged: https://github.com/encode/httpx/pull/2097
